execution/executor/src/block_executor.rs

```diff
@@ -166,6 +166,7 @@ where
         let parent_output = &parent_block.output;
         let parent_view = &parent_output.result_view;
         let parent_accumulator = parent_view.txn_accumulator();
+        parent_block.ensure_has_child(block_id)?;
```

Comment:
I don't think you can do this here, it'll fail if the block_id is a new block that's not inserted, this will fail. you need to add this in the scope of line 172 (the retry case)